src/cowboy_rest.erl

```diff
@@ -324,29 +324,33 @@ known_methods(Req, State=#state{method=Method}) ->
 uri_too_long(Req, State) ->
 	expect(Req, State, uri_too_long, false, fun allowed_methods/2, 414).
 
+stringify_allowed_methods(MethodList) when is_list(MethodList) ->
+	<< ", ", Allow/binary >> = << << ", ", M/binary >> || M <- MethodList >>,
+	Allow.
+
 %% allowed_methods/2 should return a list of binary methods.
 allowed_methods(Req, State=#state{method=Method}) ->
+	DefaultAllowedMethods = [<<"HEAD">>, <<"GET">>, <<"OPTIONS">>],
 	case call(Req, State, allowed_methods) of
 		no_call when Method =:= <<"HEAD">>; Method =:= <<"GET">> ->
-			next(Req, State, fun malformed_request/2);
+			Allow = stringify_allowed_methods(DefaultAllowedMethods),
+			Req2 = cowboy_req:set_resp_header(<<"allow">>, Allow, Req),
+			next(Req2, State, fun malformed_request/2);
 		no_call when Method =:= <<"OPTIONS">> ->
```

Comment:
We should always set the header in this function. I'm not sure it would be set in all cases for OPTIONS otherwise. Make a `set_allow_header` common function that sets the header from here, and remove the header setting from further down the code.

Reply:
So, the responsibility of adding the `allow` header will be centralized in this function, right? Sounds good to me.

Reply:
Of adding the default allow header yes. Then handlers can always override this by setting their own.

Reply:
Done!